build.sbt

```diff
@@ -159,7 +159,16 @@ lazy val lib = (project in file("lib"))
     defaultSettingsWithPlugin,
     name := "SpinalHDL-lib",
     libraryDependencies += "commons-io" % "commons-io" % "2.11.0",
+    libraryDependencies ++= Seq(
+      "com.fasterxml.jackson.core" % "jackson-core" % "2.15.3",
+      "com.fasterxml.jackson.core" % "jackson-annotations" % "2.15.3",
+      "com.fasterxml.jackson.core" % "jackson-databind" % "2.15.3",
+      "com.fasterxml.jackson.dataformat" % "jackson-dataformat-yaml" % "2.15.3",
+      "com.fasterxml.jackson.module" % "jackson-module-scala_2.12" % "2.15.3"
```

Comment:
jackson-module-scala is no more supported at 2012. I tested the old version,  it fails to serialize yaml.

Reply:
for one example:
I want to write a module which may be used by others in fusesoc or I just don't want to write some eda scripts:
``` scala
object GenerateFuseSocScript extends App {
  // it must be serializable
  case class Parameter(width: Int = 8, ports: Int = 2)
  val fusesoc = new FuseSocGeneratorBuilder[Parameter,MyTopLevel]("MyTopLevel", Parameter() ) {
    override def buildComponent(parameter: Parameter): MyTopLevel = {
      MyTopLevel(parameter.width, parameter.ports)
    }
  }
  println(fusesoc.buildScript)
}
object FuseSocEntryFunction extends App {
    ...
    fusesoc.run(args)
}
```
then I get bunch of thing like below

```yaml
CAPI=2:

name: "::MyTopLevel:0.0.0"
filesets:
  base:
    depend:
    - "chenbosoft:utils:generators:0.0.0"
generate:
  MyTopLevel_gen:
    generator: "spinalhdl"
    parameters:
      spinal_parameter:
        width: 8
        ports: 2
      output:
        files:
        - MyTopLevel.v:
            file_type: "verilogSource"
      target_directory: "./generte"
      entry_function: ""
      copy_core: false
      spinal_project_path: "."

targets:
  lint:
    default_tool : verilator
    generate: [MyTopLevel_gen]
    filesets: [base]
    tools:
      verilator:
        mode : lint-only
    toplevel : MyTopLevel
```
someone who want to use this module will copy this and modify some parameters(for example, I modify spinal_project_path  to the dir which the build.sbt located, and make entry_function=xxx.FuseSocEntryFunction)

and use `fusesoc library add spinal_generator https://github.com/chenbo-again/spinalhdl_fusesoc_ generator` to get the script(it parse yaml and pass component parameters to FuseSocEntryFunction to build and generate verilog)

finally, let the code into eda. for lint, I provide a example verilator lint, `fusesoc soc target=lint MyTopLevel`.

It's a bit complex to use fusesoc, but it can simplify many tedious tasks.

Reply:
> jackson-module-scala is no more supported at 2012. I tested the old version, it fails to serialize yaml.

may be I can try other lib?